wire-runtime/src/main/java/com/squareup/wire/RuntimeMessageAdapter.java

```diff
@@ -165,31 +165,22 @@ B newBuilder(M value) {
   }
 
   @Override public String toString(M message) {
-    StringBuilder sb = new StringBuilder();
-    sb.append(messageType.getSimpleName());
-    sb.append('{');
-    boolean seenValue = false;
+    String name = messageType.getSimpleName();
+    StringBuilder sb = new StringBuilder(name.substring(0, name.length() - 1));
     for (FieldBinding<M, B> fieldBinding : fieldBindings.values()) {
       Object value = fieldBinding.get(message);
-      if (value == null) continue;
-      if (seenValue) sb.append(", ");
-      sb.append(fieldBinding.name)
-          .append('=')
-          .append(fieldBinding.redacted ? "██" : value);
-      seenValue = true;
-    }
-    for (Extension<?, ?> extension : message.tagMap.extensions(true)) {
-      if (seenValue) sb.append(", ");
-      if (extension.isUnknown()) {
-        sb.append(extension.getTag());
-      } else {
-        sb.append(extension.getName());
+      if (value != null) {
+        sb.append(", ")
+            .append(fieldBinding.name)
+            .append('=')
+            .append(fieldBinding.redacted ? "██" : value);
```

Comment:
Define a constant for the redacted string with a comment as to what char is being used.